### PR TITLE
Add onboarding document for 1-click release process

### LINKED ISSUE #2493: Documentation meta issue
As in case of any project documentation is critical and should be added as well as maintained as the project progress. Listing few points that needs to be added as a part of the documentation: 

- [ ] Onboarding to universal release mechanism
- [ ] How remote repositories should add ci checks for their JenkinsFiles
- [ ]  Determine the smallest size a workflow can be to limit the number of practices that need to be enforced
- [ ] https://github.com/opensearch-project/opensearch-build/issues/1607
- [ ] Component: Each component should have a [RELEASING.md](https://github.com/opensearch-project/opensearch-java/blob/main/RELEASING.md) file documenting how to release the artifact
- [ ] Release system: The opensearch-build repo should have details about developing and working on jenkinsFiles and libraries. To be added in [DEVELOPER_GUIDE.md](https://github.com/opensearch-project/opensearch-build/blob/main/DEVELOPER_GUIDE.md)

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
This document adds the instruction on how to onboard to 1-click release process. 

### Issues Resolved
closes #2493 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ONBOARDING.md [hunk 1]
```diff
@@ -42,4 +43,38 @@ Add the new plugin to the [opensearch-plugins meta](https://github.com/opensearc
...
+    2. Content type: application/json
+    3. SSL verification: Enable
+    4. Which events would you like to trigger this webhook? Select `Releases` only. Please ensure to deselect `Push`
+    5. Active: Enable
+7. Once a webhook is added, it will send a ping to check the connectivity (✅)
+8. Add `RELEASING.md` file to the repository documenting how to release the artifact. [Example](https://github.com/opensearch-project/opensearch-py-ml/blob/main/RELEASING.md)
+9. **Adding tests:** Each library has a respective library tester associated with it that can be used to test you jenkins workflow. This tests can be used to verify that the workflow is making the calls. The build system used is gradle. For example, this [PublishToNpm test](https://github.com/opensearch-project/opensearch-build-libraries/blob/main/tests/jenkins/TestPublishToNpm.groovy) uses [PublishToNpmLibTester](https://github.com/opensearch-project/opensearch-build-libraries/blob/main/tests/jenkins/lib-testers/PublishToNpmLibTester.groovy) with expected parameter that can be unique to your workflow. The assertions makes sure that calls to npm registry is made which is mandatory to release an artifact.
+
```
Comment (bbarani) [inline ONBOARDING.md:79]: Overall, the on-boarding guide looks good. Can you also add an optional step for 2PR approval process ( if any repo owner is interested in using it)?

Comment (gaiksaya) [reply ONBOARDING.md:79]: Thanks for reminding. Let me add that with an example.

Comment (gaiksaya) [reply ONBOARDING.md]: Added with an example. Thanks!

**Review by bbarani — APPROVED** [review]

File: ONBOARDING.md [hunk 2]
```diff
@@ -42,4 +43,36 @@ Add the new plugin to the [opensearch-plugins meta](https://github.com/opensearc
...
+    1. It supports two test configs - `with-security` and `without-security`, which runs test with security plugin enabled and disabled respectively. Choose one or both depending on what your plugin integration tests support.
+
+
+## Onboarding to universal / 1-click release process:
+
+This document describes steps to onboard a new component to universal or 1-click release process.
+
+See https://github.com/opensearch-project/opensearch-build/issues/1234 for details about end to end workflow.
+
+1. Please ensure that [opensearch-ci-bot](https://github.com/opensearch-ci-bot) has the write access to your repository.
```
Comment (peterzhuamazon) [inline ONBOARDING.md:55]: Add sentence on if the bot does not exist, contact OSEE team.

File: ONBOARDING.md [hunk 3]
```diff
@@ -42,4 +43,36 @@ Add the new plugin to the [opensearch-plugins meta](https://github.com/opensearc
...
+1. For publishing to a new platform (other than ones specified above) a new library needs to be added. (ETA: 2 weeks)
+1. **Release Drafter**: Release drafter is a GitHub Action workflow that drafts a release that may or may not contain the release artifacts. The drafted release acts as a trigger to the Jenkins workflow. It also acts as a staging environment for release artifacts. This is to make sure the build environment remains the same even for release artifacts. [Example](https://github.com/opensearch-project/opensearch-py/blob/main/.github/workflows/release-drafter.yml)
+_**2 Person Review** It is highly recommended to add 2PR approval for any release workflow. In universal release process this can be added to release-drafter workflow as that is the starting point to trigger any release. See how to [add the mechanism in the workflow](https://github.com/opensearch-project/opensearch-dsl-py/pull/102). The mentioned solution creates an issues that notifies and assignes the reviewers. [Example](https://github.com/gaiksaya/opensearch-dsl-py/issues/6)_
```
Comment (peterzhuamazon) [inline ONBOARDING.md:60]: Additional _?

File: ONBOARDING.md [hunk 4]
```diff
@@ -42,4 +43,36 @@ Add the new plugin to the [opensearch-plugins meta](https://github.com/opensearc
...
+       - _SCM_: Github repository link. eg: https://github.com/opensearch-project/opensearch-build
+       - _Script_ Path: Relative path to jenkins file. eg: jenkins/check-for-build.jenkinsfile
+    6. Run the workflow once in order to update the configuration of the Jenkins Workflow. You can abort once the workflow starts pulling the docker image. 
+1. **GitHub Webhook**: Add webhook to your GitHub repository by going to repository settings → Webhooks:
+    - _Payload URL_: https://build.ci.opensearch.org/generic-webhook-trigger/invoke?token=\<tokenAddedInStep2\>
+    - _Content type_: application/json
+    - _SSL verification_: Enable
+    - _Which events would you like to trigger this webhook?_ Select `Releases` only. Please ensure to deselect `Push`
+    - _Active_: Enable
+1. Once a webhook is added, it will send a ping to check the connectivity (✅)
```
Comment (peterzhuamazon) [inline ONBOARDING.md:76]: More details on how to check?

File: ONBOARDING.md (hunk 2 above)
Comment (gaiksaya) [reply ONBOARDING.md:55]: Team are being removed. Will add maintainer of this repo. Thanks!

Comment (peterzhuamazon) [reply ONBOARDING.md:55]: Or link code owner file is better?

**Comment by gaiksaya** [comment]
Hi @peterzhuamazon Added the fixes. Can you review again? Thanks!

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**